sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+    std::string IfModifiedSince = std::string();  // Specify this header value to operate only on a blob if it has been modified since the specified date/time.
+    std::string IfUnmodifiedSince = std::string();  // Specify this header value to operate only on a blob if it has not been modified since the specified date/time.
+  };
+
+  static Azure::Core::Http::Request PathLeaseCreateRequest(std::string url, const PathLeaseOptions& pathLeaseOptions)
+  {
+    Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, url);
+    if (!pathLeaseOptions.ClientRequestId.empty())
+    {
+      request.AddHeader(k_HeaderClientRequestId, pathLeaseOptions.ClientRequestId);
+    }  // TODO: This should be changed to null check when Optional<T> is ready.
+    if (!std::to_string(pathLeaseOptions.Timeout).empty())
+    {
+      request.AddQueryParameter(k_QueryTimeout, std::to_string(pathLeaseOptions.Timeout));
+    }
+    request.AddHeader(k_HeaderApiVersionParameter, pathLeaseOptions.ApiVersionParameter);
+    request.AddHeader(k_HeaderPathLeaseAction, PathLeaseActionToString(pathLeaseOptions.XMsLeaseAction));
+    // TODO: This should be changed to null check when Optional<T> is ready.
+    if (!std::to_string(pathLeaseOptions.XMsLeaseDuration).empty())
+    {
+      request.AddHeader(k_HeaderXMsLeaseDuration, std::to_string(pathLeaseOptions.XMsLeaseDuration));
+    }
+    // TODO: This should be changed to null check when Optional<T> is ready.
+    if (!std::to_string(pathLeaseOptions.XMsLeaseBreakPeriod).empty())
+    {
+      request.AddHeader(k_HeaderXMsLeaseBreakPeriod, std::to_string(pathLeaseOptions.XMsLeaseBreakPeriod));
+    }
+    if (!pathLeaseOptions.LeaseIdOptional.empty())
+    {
+      request.AddHeader(k_HeaderLeaseIdOptional, pathLeaseOptions.LeaseIdOptional);
+    }
+    if (!pathLeaseOptions.ProposedLeaseIdOptional.empty())
+    {
+      request.AddHeader(k_HeaderProposedLeaseIdOptional, pathLeaseOptions.ProposedLeaseIdOptional);
+    }
+    if (!pathLeaseOptions.IfMatch.empty())
+    {
+      request.AddHeader(k_HeaderIfMatch, pathLeaseOptions.IfMatch);
+    }
+    if (!pathLeaseOptions.IfNoneMatch.empty())
+    {
+      request.AddHeader(k_HeaderIfNoneMatch, pathLeaseOptions.IfNoneMatch);
+    }
+    if (!pathLeaseOptions.IfModifiedSince.empty())
+    {
+      request.AddHeader(k_HeaderIfModifiedSince, pathLeaseOptions.IfModifiedSince);
+    }
+    if (!pathLeaseOptions.IfUnmodifiedSince.empty())
+    {
+      request.AddHeader(k_HeaderIfUnmodifiedSince, pathLeaseOptions.IfUnmodifiedSince);
+    }
+    return request;
+  }
+
+  struct PathLeaseResponse
+  {
+    std::string Date;
+    std::string ETag;
+    std::string LastModified;
+    std::string RequestId;
+    std::string Version;
+    std::string LeaseId;
+    std::string LeaseTime;
+  };
+
+  static PathLeaseResponse PathLeaseParseResponse(/*const*/ std::unique_ptr<Azure::Core::Http::Response>& responsePtr)
```

Comment:
there seems to be no need to pass pointer here, can we just pass `Azure::Core::Http::Response` by reference?

Reply:
`std::unique_ptr<Response> CurlTransport::Send(Context& context, Request& request)`
Am I missing anything here?

Reply:
I mean, the first thing you do in `XXParseResponse` function is to dereference that `std::unique_ptr`. Why not dereference it before passing it to this function?

Reply:
this is still under discussion. don't resolve unresolved conversation

Reply:
This is for future code that would save this unique_ptr of response type.